cli/src/pcluster/models/imagebuilder.py

```diff
@@ -146,6 +157,91 @@ def config(self):
             self.__config = ImageBuilderSchema().load(self.source_config)
         return self.__config
 
+    @property
+    def bucket(self):
+        """Return a bucket configuration."""
+        if self.__bucket is None:
+            self.__bucket = S3BucketFactory.init_s3_bucket(
+                service_name=self.image_name,
+                stack_name=self.image_name,
+                custom_s3_bucket=self._get_custom_bucket(),
+                artifact_directory=self.s3_artifacts_dir,
+            )
+        return self.__bucket
+
+    def _get_custom_bucket(self):
+        """
+        Try to get custom bucket name from image tag or stack tag.
+
+        If image and stack don't exist, get custom bucket name from config.
+        """
+        try:
+            image_info = AWSApi.instance().ec2.describe_image_by_name_tag(self.image_name)
+            custom_bucket_name = search_tag(image_info, PCLUSTER_S3_BUCKET_TAG)
+            if custom_bucket_name == S3Bucket.get_bucket_name(AWSApi.instance().sts.get_account_id(), get_region()):
+                custom_bucket_name = None
+        except AWSClientError as e:
+            if not isinstance(e, ImageNotFoundError):
+                raise ImageBuilderActionError(f"Unable to get bucket name from image {self.image_name} tag. {e}")
+            custom_bucket_name = None
+
+        if custom_bucket_name is None:
+            try:
+                stack_info = AWSApi.instance().cfn.describe_stack(self.image_name)
+                custom_bucket_name = search_tag(stack_info, PCLUSTER_S3_BUCKET_TAG)
+                if custom_bucket_name == S3Bucket.get_bucket_name(AWSApi.instance().sts.get_account_id(), get_region()):
+                    custom_bucket_name = None
+            except AWSClientError as e:
+                if f"Stack with id {self.image_name} does not exist" in str(e):
+                    custom_bucket_name = self.config.custom_s3_bucket
+                else:
+                    raise ImageBuilderActionError(f"Unable to get bucket name from stack {self.stack.name} tag. {e}")
+
+        return custom_bucket_name
+
+    def _get_artifact_dir(self):
+        """Get artifact directory from image tag or stack tag."""
+        try:
+            image_info = AWSApi.instance().ec2.describe_image_by_name_tag(self.image_name)
+            self.__s3_artifact_dir = search_tag(image_info, PCLUSTER_S3_IMAGE_DIR_TAG)
+            if self.__s3_artifact_dir is None:
+                raise ImageNotFoundError
+        except AWSClientError as e:
+            if not isinstance(e, ImageNotFoundError):
+                LOGGER.error("Error occurs during searching the artifact directory in image tag.")
```

Comment:
nit: let's find a more meaningful error message

Reply:
What about "Unable to get artifact directory from image {self.image_name} tag."?

Reply:
or 'Unable to find tag X in image Y'?